File: 12-integer-to-roman/integer-to-roman.js
/**
 * @param {number} num
 * @return {string}
 */
var intToRoman = function(num) {
     const valueMap = [
        [1000, "M"],[900, "CM"],[500, "D"],[400, "CD"],[100, "C"],[90, "XC"],
        [50, "L"],[40, "XL"],[10, "X"],[9, "IX"],[5, "V"],[4, "IV"],[1, "I"],
    ]
    let res = "";

    for(let [int , value] of valueMap){
        if(num === 0) break;
        const count = Math.floor(num / int);
        res += value.repeat(count);
        num -= count * int;
    }

    return res;
};

File: 12-integer-to-roman/integer-to-roman.java
class Solution {
    public String intToRoman(int num) {
        int[] tokens = {1000, 900, 500, 400, 100, 90, 
                        50, 40, 10, 9, 5, 4, 1};
        String[] values = {"M", "CM", "D", "CD", "C", "XC", 
                            "L", "XL", "X", "IX", "V", "IV", "I"};
        String ans = "";
        for(int i = 0; i < tokens.length; i ++ ) {
            if(num == 0) break;
            int count = num / tokens[i];
            num -= count * tokens[i];
            while(count > 0) {
                ans += values[i];
                count --;
            }
        }
        return ans;
    }
}
